subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ValidationActions.java

```diff
@@ -110,9 +116,81 @@ public void doValidate(String propertyName, Object values, TaskValidationContext
         }
     };
 
-    private static void validateNotInReservedFileSystemLocation(TaskValidationContext context, File location) {
+    private static void reportMissingInput(TaskValidationContext context, String kind, String propertyName, File input) {
+        context.visitPropertyProblem(problem -> {
+            String lowerKind = kind.toLowerCase();
+            problem.withId(ValidationProblemId.INPUT_FILE_DOES_NOT_EXIST)
+                .forProperty(propertyName)
+                .reportAs(ERROR)
+                .withDescription(() -> "specifies " + lowerKind + " '" + input + "' which doesn't exist")
+                .happensBecause("An input file was expected to be present but it doesn't exist")
+                .addPossibleSolution(() -> "Make sure the " + lowerKind + " exists before the task is called")
```

Comment:
`before the task is called` -> `before the task is executed`

I don't think we call tasks.